packages/core-data/src/entities.js

```diff
@@ -400,6 +374,57 @@ async function loadTaxonomyEntities() {
 	} );
 }
 
+/**
+ * Returns the Site entity.
+ *
+ * @return {Promise} Entity promise
+ */
+async function loadSiteEntity() {
+	const entity = {
+		label: __( 'Site' ),
+		name: 'site',
+		kind: 'root',
+		baseURL: '/wp/v2/settings',
+		syncConfig: {
+			fetch: async () => {
+				return apiFetch( { path: '/wp/v2/settings' } );
+			},
+			applyChangesToDoc: ( doc, changes ) => {
+				const document = doc.getMap( 'document' );
+				Object.entries( changes ).forEach( ( [ key, value ] ) => {
+					if ( document.get( key ) !== value ) {
+						document.set( key, value );
+					}
+				} );
+			},
+			fromCRDTDoc: ( doc ) => {
+				return doc.getMap( 'document' ).toJSON();
+			},
+		},
+		syncObjectType: 'root/site',
+		getSyncObjectId: () => 'index',
+		meta: {},
+	};
+
+	// @todo: Check why this isn't picking up preloaded data.
+	const site = await apiFetch( {
+		path: '/wp/v2/settings',
+		method: 'OPTIONS',
+	} );
+
+	const labels = {};
+	Object.entries( site?.schema?.properties ?? {} ).forEach(
+		( [ key, value ] ) => {
+			// Ignore properties `title` and `type` keys.
+			if ( typeof value === 'object' && value.title ) {
+				labels[ key ] = value.title;
+			}
+		}
+	);
+
+	return [ { ...entity, meta: { labels } } ];
```

Comment:
Meta 👍🏻

Reply:
Is this something that we want to normalize. Do we expect all entities to define the labels of their properties in a "meta" property. I'm guessing we need to make sure this is not already used in some endpoint.

Reply:
I see `meta` as a way to store any additional information for an entity, property label maps, and maybe perPage/maxPages info.

> I'm guessing we need to make sure this is not already used in some endpoint.

Since this is part of the configuration, I don't expect a name clash, but I'll double-check.